capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module

```diff
@@ -213,44 +213,135 @@ function c4m_features_og_events_entity_info_alter(&$entity_info) {
 }
 
 /**
- * Fetches upcoming events (Events in the future that haven't started yet),
- * Depending on the user's role (guest, logged-in),
- * This function fetches the events from user's groups or public groups.
- * 1 Event for anonymous user, 3 for logged in.
+ * Get rendered events for loggedin user.
  *
  * @return string
- *  HTML of rendered upcoming events.
+ *  HTML of rendered upcoming events, FALSE if there's no events.
  */
-function c4m_features_og_events_upcoming_events() {
-  global $user;
+function c4m_features_og_events_upcoming_events_loggedin() {
   $events = '';
 
+  // Get content filter.
+  $filter = !empty($_GET['filter']) ? $_GET['filter'] : '';
+  $range = 3;
+
+  $groups = c4m_features_og_events_get_groups_by_filter($filter);
```

Comment:
The filtering system is wrong.

The events should **not be filtered by the groups** that have topics the user is interested in.

The events should **be filtered by the events** that have topics the user is interested in.